lib/network.py

```diff
@@ -1933,6 +1939,23 @@ def get_proxies(self):
             return proxies
         return None
 
+    def on_bad_certificate(self, server, certificate):
+        if server in self.bad_certificate_servers:
+            return
+        self.bad_certificate_servers[server] = certificate
+        self.server_list_updated()
+
+    def remove_bad_certificate(self, server):
+        if not server in self.bad_certificate_servers:
+            return
+        del self.bad_certificate_servers[server]
+        self.server_list_updated()
+
+    def remove_pinned_certificate(self, server):
+        os.unlink(self.bad_certificate_servers[server])
```

Comment:
I'm just worried about potential Exceptions here if the `server` doesn't exist in the dict.. mind guarding this like the other methods are guarded above?